### PR TITLE
Reworking attack() -> use_on_mob()

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Renamed `attack()` to `use_on_mob()` and removed the zone parameter.
- Standardize parameters to `mob/living/target` and `mob/living/user`.
- Add return values and generally clean up the procs.

## Why and what will this PR improve
Standardized return value, standard parameters, less carbon types.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/game/objects/items/weapons/material/knives.dm [hunk 1]
```diff
@@ -37,16 +37,12 @@
...
-		if(user.get_target_zone() == BP_EYES)
-			if((MUTATION_CLUMSY in user.mutations) && prob(50))
-				M = user
-			return eyestab(M, user)
+/obj/item/knife/use_on_mob(mob/living/target, mob/living/user, animate = TRUE)
 
+	if(user.a_intent != I_HELP && user.get_target_zone() == BP_EYES)
+		if((MUTATION_CLUMSY in user.mutations) && prob(50))
+			target = user
+		return eyestab(target, user)
```
Comment (out-of-phaze) [inline code/game/objects/items/weapons/material/knives.dm:45]: this proc can return null, if you wanna enforce return type on `use_on_mob` you'll want either `eyestab()` to return `TRUE`/`FALSE` or just call `eyestab()` and `return TRUE` always

File: code/game/objects/items/weapons/tools/screwdriver.dm [hunk 2]
```diff
@@ -39,14 +39,12 @@
...
-		return ..()
-	if(user.get_target_zone() != BP_EYES && user.get_target_zone() != BP_HEAD)
+/obj/item/screwdriver/use_on_mob(mob/living/target, mob/living/user, animate = TRUE)
+	if(user.a_intent == I_HELP || user.get_target_zone() != BP_EYES && user.get_target_zone() != BP_HEAD)
 		return ..()
 	if((MUTATION_CLUMSY in user.mutations) && prob(50))
-		M = user
-	return eyestab(M,user)
+		target = user
+	return eyestab(target, user)
```
Comment (out-of-phaze) [inline code/game/objects/items/weapons/tools/screwdriver.dm:47]: as with knife, eyestab returns null, may wanna handle that

File: code/game/objects/items/weapons/towels.dm [hunk 3]
```diff
@@ -65,7 +65,7 @@
 		lay_out()
 		return TRUE
 	if(user.a_intent != I_HURT)
-		return attack(user, user, user.get_target_zone())
+		return use_on_mob(user, user, user.get_target_zone())
```
Comment (out-of-phaze) [inline code/game/objects/items/weapons/towels.dm]:
```suggestion
		return use_on_mob(user, user)
```

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/material/knives.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/items/weapons/material/knives.dm:45]: Went with the former.

File: code/_onclick/item_attack.dm [hunk 4]
```diff
@@ -10,7 +10,7 @@ item/resolve_attackby() calls the target atom's attackby() proc.
 Mobs:
 
 mob/living/attackby() after checking for surgery, calls the item's attack() proc.
```
Comment (out-of-phaze) [inline code/_onclick/item_attack.dm]:
```suggestion
mob/living/attackby() after checking for surgery, calls the item's use_on_mob() proc.
```

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
quick one

**Review by out-of-phaze — DISMISSED** [review]

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**